### PR TITLE
[Dev] Create canonical dev conda environment.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by frozenbugs** [comment]
@Rhett-Ying 
I am considering move CI conda and this template to dgl/conda/ dir, like:
conda/
conda/ci/...
conda/dev/...

File: script/create_dev_conda_env.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+CUDA_VERSIONS="10.2,11.3,11.6,11.7"
+TORCH_VERSION="1.12.0"
```
Comment (Rhett-Ying) [inline script/create_dev_conda_env.sh:4]: `1.12.0` or `1.12`? The latest of PyTorch 1.12 is `1.12.1`. If we use `1.12` here, the latest one will be installed?

**Review by Rhett-Ying — APPROVED** [review]

**Comment by Rhett-Ying** [comment]
add a `README.md` alongside the script to tell user how to use it?

File: script/create_dev_conda_env.sh (hunk 1 above)
Comment (frozenbugs) [reply script/create_dev_conda_env.sh:4]: Currently we are using 1.12.0 in our CI, let's update it later.

**Comment by frozenbugs** [comment]
> add a `README.md` alongside the script to tell user how to use it?

script/create_dev_conda_env.sh -h will print the usage.

**Pull request merged**